File: shared-schemas/python/sos_event.py
from enum import Enum

from pydantic import BaseModel


class SOSStatus(str, Enum):
    TRIGGERED = "triggered"
    DISPATCHED = "dispatched"
    ACKNOWLEDGED = "acknowledged"
    RESOLVED = "resolved"


class TriggerType(str, Enum):
    MANUAL = "manual"
    VOICE = "voice"
    SHAKE = "shake"


class GeoLocation(BaseModel):
    latitude: float
    longitude: float
    accuracy_meters: float | None = None


class SOSTriggerRequest(BaseModel):
    user_id: str
    location: GeoLocation
    trigger_type: TriggerType = TriggerType.MANUAL
    message: str | None = None


class SOSTriggerResponse(BaseModel):
    event_id: str
    status: SOSStatus

File: shared-schemas/src/sos-event.ts
export type SOSStatus = 'triggered' | 'dispatched' | 'acknowledged' | 'resolved';
export type TriggerType = 'manual' | 'voice' | 'shake' | 'voice_background';

export interface GeoLocation {
  latitude: number;
  longitude: number;
  accuracyMeters?: number;
}

export interface SOSTriggerRequest {
  userId: string;
  location: GeoLocation;
  triggerType: TriggerType;
  message?: string;
}

export interface SOSTriggerResponse {
  eventId: string;
  status: SOSStatus;
}

export interface SOSEventEvidence {
  type: 'audio' | 'video' | 'photo';
  url: string;
  filename: string;
  createdAt: number;
}

export interface SOSEvent {
  id: string;
  userId: string;
  location: GeoLocation;
  triggerType: TriggerType;
  status: SOSStatus;
  message?: string;
  createdAt: number;
  updatedAt: number;
  evidence?: SOSEventEvidence[];
}
